src/main/java/nixy/Nixy.java

```diff
@@ -0,0 +1,125 @@
+package nixy;
+
+import nixy.exceptions.NixyException;
+import nixy.parse.Parsed;
+import nixy.parse.Parser;
+import nixy.task.Task;
+import nixy.task.TaskList;
+import nixy.ui.Ui;
+
+/**
+ * Todo list application that allows users to manage tasks.
+ * Iterative project for CS2103
+ */
+public class Nixy {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Nixy.
+     * Initializes the user interface, storage, and task list.
+     *
+     * @param filePath The file path to store tasks data.
+     */
+    public Nixy(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (NixyException e) {
+            ui.showNixyException(e);
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Main driver for Nixy
+     * Entry point to display welcome message and read user input.
+     */
+    public void run() {
+        ui.showWelcome();
+        Boolean isExit = false;
+        while (!isExit) {
+            String input = ui.readInput();
+            Parsed p;
+            Command c;
+            try {
+                p = Parser.parse(input);
+                c = p.getCommand();
+            } catch (NixyException e) {
+                ui.showNixyException(e);
+                continue;
+            }
+            switch (c) {
```

Comment:
Perhaps it would be better to have the switch cases in another class/method?

Reply:
Perhaps I'll implement that along with the GUI, thanks for the suggestion!